spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/MultiMethodRabbitListenerEndpoint.java

```diff
@@ -47,7 +52,33 @@ protected HandlerAdapter configureListenerAdapter(MessagingMessageListenerAdapte
 			invocableHandlerMethods.add(getMessageHandlerMethodFactory()
 					.createInvocableHandlerMethod(getBean(), method));
 		}
-		return new HandlerAdapter(new DelegatingInvocableHandler(invocableHandlerMethods, this.bean));
+		this.delegatingHandler = new DelegatingInvocableHandler(invocableHandlerMethods, this.bean, getResolver(),
+				getBeanExpressionContext());
+		return new HandlerAdapter(this.delegatingHandler);
+	}
+
+
+	@Override
+	protected MessagingMessageListenerAdapter createMessageListenerInstance() {
+		return new MultiMethodMessageListenerAdapter();
+	}
+
+
+	private final class MultiMethodMessageListenerAdapter extends MessagingMessageListenerAdapter {
+
+		@Override
+		protected Address getReplyToAddress(Message request) throws Exception {
+			Address replyTo = request.getMessageProperties().getReplyToAddress();
+			Address defaultReplyTo = delegatingHandler.getDefaultReplyTo();
```

Comment:
The solution looks weak.
I understand that all these classes are the part of the infrastructure and everything is called in the right order, but this filed can be `null`, if we don't call `configureListenerAdapter`